api/trace/testtrace/span_test.go

```diff
@@ -0,0 +1,558 @@
+
+			for _, attr := range event2Attributes {
+				e.Expect(event2.Attributes[attr.Key]).ToEqual(attr.Value)
+			}
+		})
+
+		t.Run("cannot be changed after the span has been ended", func(t *testing.T) {
+			t.Parallel()
+
+			e := matchers.NewExpecter(t)
+
+			tracer := testtrace.NewTracer()
+			_, span := tracer.Start(context.Background(), "test")
+
+			subject, ok := span.(*testtrace.Span)
+			e.Expect(ok).ToBeTrue()
+
+			_, linkedSpan := tracer.Start(context.Background(), "linked")
+			expectedLink := trace.Link{
+				SpanContext: linkedSpan.SpanContext(),
+				Attributes: []core.KeyValue{
+					core.Key("key").String("value"),
+				},
+			}
+
+			subject.AddLink(expectedLink)
+			subject.End()
+
+			_, unlinkedSpan := tracer.Start(context.Background(), "not linked")
+			unexpectedLink := trace.Link{
+				SpanContext: unlinkedSpan.SpanContext(),
+			}
+
+			subject.AddLink(unexpectedLink)
+
+			links := subject.Links()
+
+			e.Expect(len(links)).ToEqual(1)
+			e.Expect(links[expectedLink.SpanContext]).ToEqual(expectedLink.Attributes)
+		})
+
+		t.Run("cannot be changed after the span has been ended", func(t *testing.T) {
+			t.Parallel()
+
+			e := matchers.NewExpecter(t)
+
+			tracer := testtrace.NewTracer()
+			_, span := tracer.Start(context.Background(), "test")
+
+			subject, ok := span.(*testtrace.Span)
+			e.Expect(ok).ToBeTrue()
+
+			subject.AddEvent(context.Background(), "test")
+
+			e.Expect(len(subject.Events())).ToEqual(1)
+
+			expectedEvent := subject.Events()[0]
+
+			subject.End()
+			subject.AddEvent(context.Background(), "should not occur")
+
+			e.Expect(len(subject.Events())).ToEqual(1)
+			e.Expect(subject.Events()[0]).ToEqual(expectedEvent)
+		})
+	})
+
+	t.Run("#Status", func(t *testing.T) {
+		t.Run("defaults to OK", func(t *testing.T) {
+			t.Parallel()
+
+			e := matchers.NewExpecter(t)
+
+			tracer := testtrace.NewTracer()
+			_, span := tracer.Start(context.Background(), "test")
+
+			subject, ok := span.(*testtrace.Span)
+			e.Expect(ok).ToBeTrue()
+
+			e.Expect(subject.Status()).ToEqual(codes.OK)
+
+			subject.End()
+
+			e.Expect(subject.Status()).ToEqual(codes.OK)
+		})
+
+		statuses := []codes.Code{
+			codes.OK,
+			codes.Canceled,
+			codes.Unknown,
+			codes.InvalidArgument,
+			codes.DeadlineExceeded,
+			codes.NotFound,
+			codes.AlreadyExists,
+			codes.PermissionDenied,
+			codes.ResourceExhausted,
+			codes.FailedPrecondition,
+			codes.Aborted,
+			codes.OutOfRange,
+			codes.Unimplemented,
+			codes.Internal,
+			codes.Unavailable,
+			codes.DataLoss,
+			codes.Unauthenticated,
+		}
+
+		for _, status := range statuses {
+			t.Run("returns the most recently set status on the span", func(t *testing.T) {
+				t.Parallel()
+
+				e := matchers.NewExpecter(t)
+
+				tracer := testtrace.NewTracer()
+				_, span := tracer.Start(context.Background(), "test")
+
+				subject, ok := span.(*testtrace.Span)
+				e.Expect(ok).ToBeTrue()
+
+				subject.SetStatus(codes.OK)
```

Comment:
Why do we need to set `codes.OK` before?

Reply:
We don't _need_ to, but it clarifies what the prior state of the status is.

Reply:
Hmm, got it.